spinoso-string/src/impls.rs

```diff
@@ -35,48 +35,48 @@ impl fmt::Write for String {
 impl io::Write for String {
     #[inline]
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-        self.buf.write(buf)
+        self.inner.as_mut_slice().write(buf)
     }
 
     #[inline]
     fn write_all(&mut self, buf: &[u8]) -> io::Result<()> {
-        self.buf.write_all(buf)
+        self.inner.as_mut_slice().write_all(buf)
     }
 
     #[inline]
     fn write_fmt(&mut self, fmt: fmt::Arguments<'_>) -> io::Result<()> {
-        self.buf.write_fmt(fmt)
+        self.inner.as_mut_slice().write_fmt(fmt)
     }
 
     #[inline]
     fn write_vectored(&mut self, bufs: &[io::IoSlice<'_>]) -> io::Result<usize> {
-        self.buf.write_vectored(bufs)
+        self.inner.as_mut_slice().write_vectored(bufs)
     }
 
     #[inline]
     fn flush(&mut self) -> io::Result<()> {
-        self.buf.flush()
+        self.inner.as_mut_slice().flush()
```

Comment:
These want to use `as_mut_vec` to allow the String to grow.

Reply:
Alternatively, should `io::Write` be implemented on each encoded type?

Reply:
I like the sound of that. It seems consistent with the other delegation we're doing

Reply:
Super!

Also, sorry on the lack of attention to detail on my behalf in this PR. I really should have checked my changes a bit more for this sort of stuff. On the right track to splitting this all up though!

Reply:
Done!